### PR TITLE
Disable custom command groups (Suggestion #2224)

### PR DESCRIPTION
**Pull request by Borbot33**
# What's changed?

- Small warning if the command group is disabled
- A switch for disabling the custom commang group

[Suggestion #2224](https://discord.com/channels/166207328570441728/356486960417734666/1246836065872908298)

# Screenshots
image
screenshot 2

### CODE REVIEW

File: customcommands/assets/customcommands.html [hunk 1]
```diff
@@ -106,6 +106,10 @@ <h2>Custom commands</h2>
                                         {{textChannelOptionsMulti .ActiveGuild.Channels .CurrentCommandGroup.IgnoreChannels }}
                                     </select>
                                 </div>
+                                <div class="form-group">
+                                    <label>Group disabled</label><br>
+                                    {{checkbox "disabled" "disabled" "Enables or Disables the group" .CurrentCommandGroup.Disabled }}
```
Comment (jo3-l) [inline customcommands/assets/customcommands.html:111]: nit
```suggestion
                                    {{checkbox "disabled" "disabled" "Enables or Disables the group" .CurrentCommandGroup.Disabled}}
```

**Review by jo3-l — CHANGES_REQUESTED** [review]
You should load the related group then use `cmd.R.Group.Disabled` instead of adding `getDisabledGroups`, which fetches all disabled groups even if we only need information about one.

**Comment by Borbot33** [comment]
> You should load the related group then use `cmd.R.Group.Disabled` instead of adding `getDisabledGroups`, which fetches all disabled groups even if we only need information about one.

done :)

File: customcommands/web.go [hunk 2]
```diff
@@ -643,11 +644,16 @@ func handleRunCommandNow(w http.ResponseWriter, r *http.Request) (web.TemplateDa
 	}
 
 	// only interval commands can be ran from the dashboard currently
-	cmd, err := models.CustomCommands(qm.Where("guild_id = ? AND local_id = ? AND trigger_type = 5", activeGuild.ID, cmdID)).OneG(context.Background())
+	cmd, err := models.CustomCommands(qm.Where("guild_id = ? AND local_id = ? AND trigger_type = 5", activeGuild.ID, cmdID), qm.Load("Group")).OneG(context.Background())
 	if err != nil {
 		return templateData, err
 	}
 
+	if cmd.GroupID.Int64 != 0 && cmd.R.Group.Disabled {
```
Comment (jo3-l) [inline customcommands/web.go:652]: (untested)
```suggestion
	if cmd.R.Group != nil && cmd.R.Group.Disabled {
```
is probably less brittle and ditto everywhere else.

**Review by jo3-l — COMMENTED** [review]
Nice, LGTM other than one minor nit

File: customcommands/tmplextensions.go [hunk 3]
```diff
@@ -180,11 +180,20 @@ func tmplRunCC(ctx *templates.Context) interface{} {
 			return "", templates.ErrTooManyCalls
 		}
 
-		cmd, err := models.FindCustomCommandG(context.Background(), ctx.GS.ID, int64(ccID))
+		// cmd, err := models.FindCustomCommandG(context.Background(), ctx.GS.ID, int64(ccID))
+		// if err != nil {
+		// 	return "", errors.New("Couldn't find custom command")
+		// }
+
```
Comment (jo3-l) [inline customcommands/tmplextensions.go:187]: You can remove this commented-out code completely, and likewise below. Otherwise everything looks OK to me.

Comment (Borbot33) [reply customcommands/tmplextensions.go]: yup :)

**Pull request merged**